### PR TITLE
Ingress performance testing with async storage client + async statistics + uvloop(core only) + ujson

### PR DESCRIPTION
**Pull request by amarendra-dianomic**
Summary of modifications:
- Added Async Storage Client + Async Readings client in storage_client.py
- Added async statistics client vide statistics_async.py to use async Storage client
- Modified ingest.py to use async Storage client + async statistics client
- Fixed COAP incoming recs loss in ingest.py

However, we have retained sync storage and sync statistics client. These will be removed after converting all tasks/processes/schedules to use async storage + async statistics client such as:
- Purge process
- Statistics Collector
- Backup/Restore
- North tasks

### CODE REVIEW

File: python/foglamp/common/storage_client/storage_client.py [hunk 1]
```diff
@@ -7,23 +7,24 @@
...
 
 from foglamp.common import logger
 from foglamp.common.service_record import ServiceRecord
 from foglamp.common.storage_client.exceptions import *
 from foglamp.common.storage_client.utils import Utils
 
-_LOGGER = logger.setup(__name__, level=20)
-
+_LOGGER = logger.setup(__name__)
+MAX_ATTEMPTS = 5
```
Comment (praveen-garg) [inline python/foglamp/common/storage_client/storage_client.py]: NOPE!

File: python/foglamp/common/storage_client/storage_client.py [hunk 2]
```diff
@@ -635,20 +638,27 @@ def disconnect(self):
             raise TypeError("Provided data to insert must be a valid JSON")
 
         post_url = '/storage/table/{tbl_name}'.format(tbl_name=tbl_name)
-
-        try:
-            url = 'http://' + self.base_url + post_url
-            async with aiohttp.ClientSession() as session:
-                async with session.post(url, data=data) as resp:
+        attempt_count = 1
+        while True:
```
Comment (praveen-garg) [inline python/foglamp/common/storage_client/storage_client.py]: No, attempts thing are meant for caller not the storage client code.

File: python/foglamp/services/south/ingest.py [hunk 3]
```diff
@@ -205,7 +207,9 @@ class Ingest(object):
         cls._parent_service = parent
 
         cls.readings_storage = ReadingsStorageClient(cls._core_management_host, cls._core_management_port)
+        cls.readings_storage_async = ReadingsStorageClientAsync(cls._core_management_host, cls._core_management_port)
         cls.storage = StorageClient(cls._core_management_host, cls._core_management_port)
```
Comment (praveen-garg) [inline python/foglamp/services/south/ingest.py]: what is the need of this sync version now?

File: python/foglamp/services/south/ingest.py [hunk 4]
```diff
@@ -205,7 +207,9 @@ class Ingest(object):
         cls._parent_service = parent
 
         cls.readings_storage = ReadingsStorageClient(cls._core_management_host, cls._core_management_port)
```
Comment (praveen-garg) [inline python/foglamp/services/south/ingest.py]: what is the need of this sync version now?

File: python/foglamp/services/south/ingest.py [hunk 5]
```diff
@@ -499,7 +491,7 @@ def is_available(cls) -> bool:
                     cls._current_readings_list_index = list_index
                     return True
 
-        _LOGGER.warning('The ingest service is unavailable')
+        _LOGGER.warning('The ingest service is unavailable list_index: %s, list_size: %s', list_index, len(cls._readings_lists[list_index]))
```
Comment (praveen-garg) [inline python/foglamp/services/south/ingest.py]: %s or %d

File: python/requirements.txt [hunk 6]
```diff
@@ -18,3 +18,7 @@ pexpect==4.0.1
 
 # Transformation of data, Apply JqFilter
 pyjq==2.1.0
+
+# Below required for performance testing and comparison
```
Comment (praveen-garg) [inline python/requirements.txt]: This is not a valid comment !

File: python/foglamp/common/storage_client/storage_client.py (hunk 2 above)
Comment (amarendra-dianomic) [reply python/foglamp/common/storage_client/storage_client.py]: Nope. Here caller IS the async storage client method, hence this exception should be handled here only and NOT by the caller of the async storage client method and should inform the caller of the failure only when a timeout occurs.

**Comment by praveen-garg** [comment]
This has been splitted into #819 #820

**Comment by amarendra-dianomic** [comment]
Closed in favour PR #819, #820, #824 and #825.

**Pull request closed without merging**